dwave/system/composites/linear_ancilla.py

```diff
@@ -0,0 +1,214 @@
+            can be done with :class:`dimod.ScaleComposite`.
+
+        Args:
+            bqm (:obj:`~dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            h_tolerance (:class:`numbers.Number`):
+                Magnitude of the linear bias can be left on the qubit. Assumed to be positive. Defaults to zero.
+
+            default_flux_bias_range (:class:`typing.Sequence`):
+                Flux bias range safely accepted by the QPU, the larger the better.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child
+                sampler.
+
+        Returns:
+            :obj:`~dimod.SampleSet`
+
+        """
+        if h_tolerance < 0:
+            raise ValueError("h_tolerance needs to be positive or zero")
+
+        child = self.child
+        qpu_properties = innermost_child_properties(child)
+        g_target = child.to_networkx_graph()
+        g_source = dimod.to_networkx_graph(bqm)
+        j_range = qpu_properties["extended_j_range"]
+        flux_bias_range = qpu_properties.get("flux_bias_range", default_flux_bias_range)
+
+        # Positive couplings tend to have smaller control error,
+        # we default to them if they have the same magnitude than negative couplings
+        # https://docs.dwavesys.com/docs/latest/c_qpu_ice.html#overview-of-ice
+        largest_j = j_range[1] if abs(j_range[1]) >= abs(j_range[0]) else j_range[0]
+        largest_j_sign = np.sign(largest_j)
+
+        # To implement the bias sign through flux bias sign,
+        # we pick a range (magnitude) that we can sign-flip
+        fb_magnitude = min(abs(b) for b in flux_bias_range)
+        flux_biases = [0] * qpu_properties["num_qubits"]
+
+        _bqm = bqm.copy()
+        used_ancillas = defaultdict(list)
+        for variable, bias in bqm.iter_linear():
+            if abs(bias) <= h_tolerance:
+                continue
+            if abs(bias) - h_tolerance > abs(largest_j):
+                return NotImplementedError(
+                    "linear biases larger than the strongest coupling are not supported yet"
+                )  # TODO: implement larger biases through multiple ancillas
+
+            available_ancillas = set(g_target.adj[variable]) - set(g_source.nodes())
+            if not len(available_ancillas):
+                raise ValueError(f"variable {variable} has no ancillas available")
+            unused_ancillas = available_ancillas - set(used_ancillas)
+            if len(unused_ancillas):
+                ancilla = unused_ancillas.pop()
+                # bias sign is handled by the flux bias
+                flux_biases[ancilla] = np.sign(bias) * largest_j_sign * fb_magnitude
+                _bqm.add_interaction(
+                    variable, ancilla, (abs(bias) - h_tolerance) * largest_j_sign
+                )
+            else:
+                if qpu_properties["j_range"][0] <= bias <= qpu_properties["j_range"][1]:
```

Comment:
Shouldn't this be extended j range?

Reply:
No, here I'm capturing the symmetric range that can be sign flipped. Otherwise, the qubit is limited to utilizing ancillas with the appropriate sign in the flux bias